modules/web/js/ballerina/components/panel-decorator.jsx

```diff
@@ -298,18 +298,36 @@ class PanelDecorator extends React.Component {
     }
 
     onDropZoneActivate(e) {
-        const dragDropManager = this.context.dragDropManager,
-            dropTarget = this.props.dropTarget,
-            dropSourceValidateCB = this.props.dropSourceValidateCB;
+        const dragDropManager = this.context.dragDropManager;
+        const dropTarget = this.props.dropTarget;
+        const dropSourceValidateCB = this.props.dropSourceValidateCB;
+        const droppedNodeIndexCallBack = function () {
+            if (BallerinaASTFactory.isConnectorDeclaration(dragDropManager.getNodeBeingDragged())) {
```

Comment:
Here the consideration was, that the panel is the one who should decide which should be the index. Connector declaration should not position himself because based on the drop zone it is being positioned. WDYT?